esphome/components/wifi/wifi_component.h

```diff
@@ -264,6 +264,7 @@ class WiFiComponent : public Component {
 
   bool has_sta() const;
   bool has_ap() const;
+  bool is_ap_enabled() const;
```

Comment:
```suggestion
  WiFiComponentState get_state() const { return this->state_; }
```
Just stumbled upon this PR accidentally. Great work! 🙏 

I'm no maintainer and not versed into this codebase so take with a fair grain of salt, but when looking through, I wondered if, _since you already extend `wifi` component_, it wouldn't be more futureproof to just expose the internal `state_` enum instead of yet another **bool** -> it'd allow other components to get more info about the WiFi state from one place.

I.e. I believe what you need could then still be accomplished via:
``` cpp
wifi::global_wifi_component->get_state() != wifi::WIFI_COMPONENT_STATE_AP
```

Reply:
When you modify the *StatusIndicator* component as well with your suggested change, i will commit this to this PR as well.

Anyway thanks for your contributions.

Reply:
Well.. I wasn't planning to upstream my tiny changes and merely wanted to hint this for your consideration, but... on a 2nd thought... why not 😁 Here you go: [GH-391](https://github.com/nielsnl68/esphome/pull/391).
Feel free to do with it as you please (edit/discard/merge...). I don't claim any authorship over this one-liner nor am I planning to extend it further.

Note it changes the semantics slightly: from `has AP enabled` to `has **only** AP enabled`. It is how that particular `status_` enum works internally and was more useful for me this way.

Reply:
I just tried to use this on my nanoESP32-C6-N16:

```yaml
external_components:
  #- source: github://pr#5814
  #  components: [ status_indicator ]
  - source: 
      type: git
      url: https://github.com/mbronk/esphome
      ref: private/mbronk/nvds-status-info-forked
    components: [ status_indicator ]
```

But it fails compilation with:

```
src/esphome/components/status_indicator/status_indicator.cpp: In member function 'virtual void esphome::status_indicator::StatusIndicator::loop()':
src/esphome/components/status_indicator/status_indicator.cpp:75:56: error: 'class esphome::wifi::WiFiComponent' has no member named 'get_state'; did you mean 'get_sta'?
   75 |     if (status.empty() && wifi::global_wifi_component->get_state() == wifi::WIFI_COMPONENT_STATE_AP) {
      |                                                        ^~~~~~~~~
      |                                                        get_sta
```